test/sql/local/irc/test_incorrect_secret_and_attach.test

```diff
@@ -0,0 +1,70 @@
+# name: test/sql/local/irc/test_incorrect_secret_and_attach.test
+# description: test combinations of create secret and attah
+# group: [irc]
+
+require avro
+
+require parquet
+
+require iceberg
+
+require httpfs
+
+# Test catalog secret and storage secret
+statement ok
+set enable_logging=true
+
+statement ok
+set logging_level='debug'
+
+# auth url doesn't provide token, error message should be informative
+statement error
+ATTACH '' AS my_datalake (
+    TYPE ICEBERG,
+    CLIENT_ID 'admin',
+    CLIENT_SECRET 'password',
+    OAUTH2_SERVER_URI 'http://127.0.0.1:8181/bad_oauth_url',
+    ENDPOINT 'http://127.0.0.1:8181'
+);
+----
+Error: Could not get token from http://127.0.0.1:8181/bad_oauth_url
+
+# check the POST request to the URL
+query I
+SELECT message FROM duckdb_logs where type='iceberg.Catalog.Curl.HTTPRequest' and message like '%POST%' order by timestamp
+----
+POST http://127.0.0.1:8181/bad_oauth_url (curl code 'No error')
+
+# we don't allow oauth2 without oauth2_server_uri
+statement error
+CREATE SECRET iceberg_secret (
+	TYPE ICEBERG,
+	CLIENT_ID 'admin',
+	CLIENT_SECRET 'password',
+	AUTHORIZATION_TYPE 'oauth2'
+);
+----
+Invalid Input Error: No 'oauth2_server_uri' was provided
```

Comment:
This error message can be the same as the one in `test_create_secret_attach_combinations.test`
```
AUTHORIZATION_TYPE is 'oauth2', yet no 'oauth2_server_uri' was provided, and no 'endpoint' was provided to fall back on. Please provide one or change the 'authorization_type'.
```